### ISSUE TITLE
Jupyter displays "Directory not found" menu permanently

### ISSUE THREAD

**Issue by pcrespov** [ITISFoundation/osparc-issues#412]
@pcrespov commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017)

POssible causes
- Garbage collector stops service?
- User downloads several GB to inputs (S3 timeout?)

_Issue created from a [Mattermost message](https://mattermost.speag.com:443/_redirect/pl/bai4583fq3ry7bixo1hwmmg9xe) *by `@newton`*._

---

@GitHK commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-737811166)

There is fix for this https://github.com/ITISFoundation/osparc-simcore/pull/1956
Please make sure it is deployed on the environment where it emerged.

---

@pcrespov commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-737826606)

> There is fix for this #1956
> Please make sure it is deployed on the environment where it emerged.

FIX #1956 is already in production. See [release notes](https://github.com/ITISFoundation/osparc-simcore/releases/tag/v1.4.0)
<img width="825" alt="Screen Shot 2020-12-03 at 11 09 56" src="https://user-images.githubusercontent.com/32402063/100995459-16bfe100-3558-11eb-81d1-c80b04a648d4.png">

---

@pcrespov commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-737834809)

- 2.2.30 is latest release in production. 
<img width="215" alt="Screen Shot 2020-12-03 at 11 15 46" src="https://user-images.githubusercontent.com/32402063/100996060-e7f63a80-3558-11eb-91aa-7faf49927be7.png">

includes changes by ANE 

<img width="474" alt="Screen Shot 2020-12-03 at 11 15 31" src="https://user-images.githubusercontent.com/32402063/100996024-ddd43c00-3558-11eb-8f0a-ea2d6afc99dc.png">

![image](https://user-images.githubusercontent.com/32402063/100996236-1d028d00-3559-11eb-94ac-cf08e1c50ea6.png)

So we can conclude that all related fixes ARE deployed

---

@GitHK commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-737840541)

I can also confirm that the heartbeat is working. As it is being sent every 2 seconds, so all is deployed.

![Screenshot 2020-12-03 at 11 21 16](https://user-images.githubusercontent.com/5694077/100996700-ac0fa500-3559-11eb-8398-6beb931bc154.png)

---

@pcrespov commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-737854986)

log [webserver around 18h yesterday](https://monitoring.osparc.speag.com/graylog/search?q=container_name%3Aproduction%5C-z43%5C-simcore_production_web%2A&rangetype=absolute&from=2020-12-02T16%3A08%3A28.931Z&to=2020-12-02T20%3A04%3A29.501Z) shows problems stopping services because service was NOT found (ServiceNotFoundError is a director-client error when failing [to stop a service](https://github.com/ITISFoundation/osparc-simcore/blob/2834fd9fc33da80ee01b0477068be46bf1892e73/services%2Fweb%2Fserver%2Fsrc%2Fsimcore_service_webserver%2Fdirector%2Fdirector_api.py#L92) )

```
WARNING[0m: servicelib.utils:logged_gather(93) - Error in 1-th concurrent task <coroutine object stop_service at 0x7f5a506ffd58>: Service with uuid 406daf04-5d80-4878-9688-c13bf1112be1 not found
```

But noticed that these services remain alive one day later ... how could the director miss them and why they were garbage collected?!

---

@pcrespov commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-738226860)

Takes 4 mins to download data… it might be that the server un-responsive!?
in a jupyter smash v2.2.30 service

---

@mguidon commented on [Thu Dec 03 2020](https://github.com/ITISFoundation/osparc-simcore/issues/2017#issuecomment-738281626)

https://github.com/jupyterlab/jupyterlab/issues/6782

Might be a similar issue since we tend to move files around in the background

**Comment by pcrespov** [comment]
Duplicated in #375

**Comment by odeimaiz** [comment]
@AntoninoMarioC:
"I opened multiple instances of the same project on different tabs."

**Comment by odeimaiz** [comment]
@newton1985:
"Every 20 minutes or so I get the "Directory "" not found" message"

**Comment by pcrespov** [comment]
### Status: 
This cause of this issue is still not clear, we have done some preventive changes but the issue might still persist

- The failure can be reproduced by accessing the container console while the service run and and ``rm -rf work``. This causes JupyterLab to trigger a modal window with ``Directory Not Found`` that cannot be removed until the folder is reestablished.

So far we have two hypothesis about the cause of this issue
- In [PR #2262](https://github.com/ITISFoundation/osparc-simcore/pull/2262) we reviewed whether our ``node_ports`` library was causing this issue and did not find any evidence of it. Nonetheless, we added some logs that monitor the state of ``work`` directory to help understand the cause that deletes the folder. 
- Another potential cause might be that the garbage collector closes the services while it is still in use. Up to now, we did not find evidence of this case.

NEXT? The new mount system and the sidecar under development should mitigate both causes above. When those are in place, we will review this issue again.

**Comment by pcrespov** [comment]
Possible causes for this error: 
  - loses connection with the associated dynamic service ( e.g. gc shut it down, crashed, ... )

Possible solutions:
- update jupyterlab: new jupyterlab versions messages better @KZzizzle @GitHK 
- front-end adds a health-check on the iframe's associated dynamic service and shows as a "light" in the UI @odeimaiz 
- new dynamic sidecar: reboot option? healthcheck? error-handling? @GitHK

**Comment by sanderegg** [comment]
@pcrespov : This message appears in the following reproducible case:
- jupyterlab is started
- the docker service of jupyterlab disappears
- the jupyterlab frontend loses the connection and receives a 404 -> Directory not found error (and the jupyterlab frontend sends million calls to the backend)

This part should be fixed by [osparc-simcore#2710](https://github.com/ITISFoundation/osparc-simcore/pull/2710)

**Comment by elisabettai** [comment]
Should be solved by the linked pull requests.

Now, in case of disconnection from the server, a more meaningful message will be shown (e.g. "Server Disconnected") instead of "Directory not found" or authentication pop-up.
